utils/src/main/kotlin/FileMatcher.kt

```diff
@@ -38,7 +38,15 @@ class FileMatcher(
         /**
          * A matcher which uses the default license file names.
          */
-        val LICENSE_FILE_MATCHER = FileMatcher(LICENSE_FILENAMES + ROOT_LICENSE_FILENAMES)
+        val LICENSE_FILE_MATCHER: FileMatcher =
+            run {
+                // The (java.nio.*) glob matcher implementation does not allow for matching "zero or multiple" trailing
+                // directories with a single glob expression. Work around this by creating two patterns per filename.
+                val filenames = LICENSE_FILENAMES + ROOT_LICENSE_FILENAMES
+                val patterns = filenames.flatMap { listOf("**/$it", it) }
+
+                FileMatcher(patterns)
```

Comment:
I wonder whether this really is a right step towards generalizing the root license handling. Because IMO you actually do *not* want to match files recursively in an unconditional way, as it plays an important role *where* (in which directory) the match was found. Instead, I would have expected a manual directory walk upwards the tree, and in each directory do a non-recursive match, and stop once a match has been found.

Reply:
In my view what you describe is not the responsibility of this class. I think the responsibility is to determine whether a given `file path` matches a `license file name`, whereas the decision only knows about that single `file path`, but not about the existence of any other files. 

IIUC what you describe refers more to the decision about which of the files which do actually match a license file name do actually apply to another filepath `y`. That other file path `y` is actually not known here.

Figuring out which license files do actually apply, we have currently amongst others in ScanResult.filterPath().
In a following PR I was planning to adjust that one to keep only the license files which do actually apply. Does this make sense to you?

Reply:
@sschuberth  need to repork the PR, I'll leave a note when it's up for review again.

Reply:
@sschuberth I've ditched the questionable bit, so that the remaining two minor change would be up for review again.